ndb/src/google/cloud/ndb/model.py

```diff
@@ -288,11 +287,32 @@ def __hash__(self):
         return hash((self.definition, self.state, self.id))
 
 
-class ModelAdapter:
-    __slots__ = ()
+def _entity_from_protobuf(protobuf):
+    """Deserialize an entity from a protobuffer.
 
-    def __init__(self, *args, **kwargs):
-        raise NotImplementedError
+    Args:
+        protobuf (google.cloud.datastore_v1.proto.entity.Entity): An
+            entity protobuf to be deserialized.
+
+    Returns:
+        .Model: The deserialized entity.
+    """
+    ds_entity = helpers.entity_from_protobuf(protobuf)
+    model_class = Model._lookup_model(ds_entity.kind)
+    entity = model_class()
+    entity.key = key_module.Key._from_ds_key(ds_entity.key)
+    for name, value in ds_entity.items():
+        prop = getattr(model_class, name, None)
+        if not (prop and isinstance(prop, Property)):
+            continue
+        if value is not None:
+            if prop._repeated:
+                value = list(map(_BaseValue, value))
```

Comment:
`list(map(...))` not so fun? Use a list comprehension instead?

Reply:
3609595